pkg/sync/sync_service.go

```diff
@@ -121,12 +121,15 @@ func (syncService *SyncService[H]) initStoreAndStartSyncer(ctx context.Context,
 	if initial.IsZero() {
 		return errors.New("failed to initialize the store and start syncer")
 	}
-	if err := syncService.store.Init(ctx, initial); err != nil {
+
+	if err := syncService.StartSyncer(ctx); err != nil {
 		return err
 	}
-	if err := syncService.StartSyncer(ctx); err != nil {
+
+	if err := syncService.store.Append(ctx, initial); err != nil {
 		return err
 	}
+
 	return nil
```

Comment:
The problem is that `Init` was idempotent. It was noop if the store was already initialized. The `Append`, on the other hand, will overwrite the existing header, so you shouldn't call it every time you start the node.

Additionally, examining LOC 142 reveals that you are actually initializing it twice. I'm not sure if that's intended, but you shouldn't be doing this either with `Append`. Only one first genesis run. Also, I bet you could rewrite this logic so that `isGenesis` disappears and genesis is handled as any other header, but that's a different story